lib/plugins.js

```diff
@@ -259,6 +259,14 @@ const _getPattern = plugin => {
   }
 };
 
-exports.isPluginFor = path => {
-  return plugin => _getPattern(plugin).test(path);
+exports.patternFor = (plugin, isStatic) => {
+  if (isStatic) {
+    return _getPattern(plugin.staticExtension || plugin.extension, plugin.staticPattern || plugin.pattern);
```

Comment:
i.e. for static compilation, we will first try `staticExtension` and `staticPattern` (optional), and only then revert to the regular `extension`/`pattern`